fe/fe-core/src/main/java/com/starrocks/catalog/OlapTable.java

```diff
@@ -1159,6 +1184,25 @@ public Map<String, Range<PartitionKey>> getValidPartitionMap(int lastPartitionNu
         return result;
     }
 
+    // partitionName -> formatValue 1:1/1:N
+    public Map<String, List<List<String>>> getValidListPartitionMap(int lastPartitionNum) {
+        Map<String, List<List<String>>> listPartitionMap = getListPartitionMap();
+        // less than 0 means not set
+        if (lastPartitionNum < 0) {
+            return listPartitionMap;
+        }
+
+        int partitionNum = listPartitionMap.size();
+        if (lastPartitionNum > partitionNum) {
+            return listPartitionMap;
+        }
+
+        return listPartitionMap.entrySet().stream()
```

Comment:
if the table has many partitions, the compareByValue may have performance problem?

Reply:
I don't know, but we can test it later and optimize it later.